### DISCUSSION TITLE
It's working!!!

### DISCUSSION THREAD

**Discussion by Witchdoc59** [PlummersSoftwareLLC/NightDriverStrip#674] [Show and tell]
I had trouble getting Dave's laser cannon working.  [https://www.youtube.com/watch?v=1hYmpztMQe0&list=PLF2KJ6Gy3cZ7ynsp8s4tnqEFmY15CKhmH]  So then I saw a video on WLED and I thought I'd try that.  WLED can not use pin 32 so I had to wire up my M5 using pin 33.    Following the wiring scheme of the WLED video everything worked fine.     So once I had WLED working I thought maybe I could get NightDriver working.   It took me a bit of time to find where to modify the active pin but once I set 33 in the globals.h file then everything worked fine.  I've attached a photo of the wiring setup and a video of the working strip.   I haven't got the strip rolled out to it's full length as the room isn't big enough.
![20241215_132436](https://github.com/user-attachments/assets/0a91ae0d-8cc8-4c43-a90e-a6fc60127b1c)

I created a video of the working strip and it can be viewed here.   https://youtu.be/dReUvsyvFTg

I'm not sure if I have the #define COLOR_ORDER EOrder::GRB set correctly.   WiFi and Remote Control are not working.   The problem with the WiFi is that it is trying to join the wrong wifi network even though I've set the correct ssid in the secrets.h file.   I'm suspecting that it's reading a wrong secrets.h file.

**Comment by rbergen** [comment]
Glad to hear you've got it working!

> WiFi and Remote Control are not working. The problem with the WiFi is that it is trying to join the wrong wifi network even though I've set the correct ssid in the secrets.h file. I'm suspecting that it's reading a wrong secrets.h file.

I actually suspect you booted the device once while the wrong SSID was still configured, at which time NDS would have persisted that into nvram. Once it finds the credentials there it will ignore the defaults in secrets.h, even if the stuff in nvram is wrong. 

You could try doing a full (firmware) reset of the device, reflash with the correct settings in secrets.h and then try again.

I don't think the remote control can work, as I don't see an IR receiver connected to your M5StickC Plus2 in the picture or video.

**Reply by Witchdoc59** [reply]
I thought the m5 had a built in IR Reciever.   

I flashed the m5 with an example from the Arduino m5 library.  RTC.ino, that worked fine so maybe now I can upload the nightdriver and it will have the correct SSID.

**Reply by Witchdoc59** [reply]
Well that worked.  At first the m5 said there was no wifi so I opened up the serial port and noticed that the serial monitor indicated that I had an IP address and then a second look at the m5 and it's showing the IP address.

**Comment by davepl** [comment]
The M5 has an LED IR transmitter, but no receiver.
